src/Shmuelie.WinRTServer/Internal/BaseClassFactoryProxy.cs

```diff
@@ -63,7 +63,7 @@ public uint Release()
     private static class Impl
     {
         [UnmanagedFunctionPointer(CallingConvention.StdCall)]
-        public delegate int QueryInterfaceDelegate(BaseClassFactoryProxy* @this, Guid* riid, void** ppvObject);
+        public delegate HRESULT QueryInterfaceDelegate(BaseClassFactoryProxy* @this, Guid* riid, void** ppvObject);
```

Comment:
The downlevel support is kinda unfortunate, you could just use `[GeneratedComClass]` otherwise?

Reply:
Investigating doing that is on my To Do, see #12